### DISCUSSION TITLE
Metrics documentation

### DISCUSSION THREAD

**Discussion by js8080** [jaegertracing/jaeger-client-csharp#206] [Q&A] [answered]
Hello, I am using the Jaeger C# Client and I would like to publicize the metrics gathered by it via Prometheus.

I've written my own IMetricsFactory, ICounter, IGauge, and ITimer classes that essentially wrap corresponding classes in the [Prometheus-Net library](https://github.com/prometheus-net/prometheus-net) but I'm having a little trouble with it. For one, I don't understand what the `Dictionary<string. string> tags` is really representing in the `IMetricsFactory.Create*` methods, for example:

```csharp
public ICounter CreateCounter(string name, Dictionary<string, string> tags)
```

In my debugging, I see tag key:value pairs such as:
```
state:joined
sampled:y
```

Should I represent these as labels in my Prometheus metrics?

It'd be nice to have more details in a README.

**Comment by yurishkuro** [comment, marked as answer]
Yes, tags are "labels" in Prometheus speak.

**Comment by Falco20019** [comment]
This is the draft that we created some time ago but have not finished:
https://github.com/jaegertracing/jaeger-client-csharp/commit/67ea8dca71571c164ebe163323de9d1ebf14c515
